libworkbench: implement GLogWriterFunc and library initialization

andyholmes opened the pull request:
Add `Workbench.init()` to initialize resources and other setup tasks.

Currently, this only implements a `GLogWriterFunc` for silencing
specific messages that Workbench users shouldn't see.

This method has some limitations running on other threads in GJS,
and should be more performant in C anyways.

sonnyp reviewed (commented):
Can you add the rational / benefit of this in the PR description?

andyholmes committed:
libworkbench: implement GLogWriterFunc and library initialization

Add `Workbench.init()` to initialize resources and other setup tasks.

Currently, this only implements a `GLogWriterFunc` for silencing
specific messages that Workbench users shouldn't see.

This method has some limitations running on other threads in GJS,
and should be more performant in C anyways.
